packages/frontend/src/components/common/ResourceView/ResourceViewGrid/ResourceViewGridSpaceItem.tsx

```diff
@@ -36,107 +54,119 @@ const ResourceViewGridSpaceItem: FC<ResourceViewGridSpaceItemProps> = ({
-        >
-            <Group p="md" align="center" spacing="md" noWrap>
-                <ResourceIcon item={item} />
-
-                <Stack spacing={4} sx={{ flexGrow: 1, flexShrink: 1 }}>
-                    <Text lineClamp={1} fz="sm" fw={600}>
-                        {item.data.name}
-                    </Text>
+    const renderAccess = useCallback(() => {
+        let Icon = IconLock;
+        let status = 'Private';
 
-                    <Group spacing="sm">
-                        <Flex align="center" gap={4}>
-                            {!item.data.isPrivate ? (
-                                <>
-                                    <IconUsers
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+        if (!item.data.isPrivate) {
+            Icon = IconUsers;
+            status = 'Public';
+        } else if (item.data.access.length > 1) {
+            Icon = IconUser;
+            status = 'Shared';
+        }
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Public
-                                    </Text>
-                                </>
-                            ) : item.data.access.length > 1 ? (
-                                <>
-                                    <IconUser
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+        return (
+            <>
+                <Icon color={theme.colors.gray[6]} size={14} />
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Shared
-                                    </Text>
-                                </>
-                            ) : (
-                                <>
-                                    <IconLock
-                                        color={theme.colors.gray[6]}
-                                        size={14}
-                                    />
+                <Text size={14} color="gray.6" fz="xs">
+                    {status}
+                </Text>
+            </>
+        );
+    }, [item]);
 
-                                    <Text size={14} color="gray.6" fz="xs">
-                                        Private
-                                    </Text>
-                                </>
-                            )}
-                        </Flex>
+    const getToolTipText = useCallback(() => {
+        if (!item.data.isPrivate) {
+            return 'Everyone in this project has access';
+        }
+        if (item.data.access.length > 1) {
+            const userCount = item.data.access.length;
+            return `Shared with ${userCount} user${userCount > 1 ? 's' : ''}`;
+        }
+        return 'Only visible to you';
+    }, [item]);
```

Comment:
same as above, getResourceAccessType can remove code duplication that happening here